src/libsyntax_ext/proc_macro_server.rs

```diff
@@ -336,12 +336,12 @@ impl Ident {
         }
     }
     fn new(sym: Symbol, is_raw: bool, span: Span) -> Ident {
-        let string = sym.as_str().get();
-        if !Self::is_valid(string) {
+        let string = sym.as_str();
+        if !Self::is_valid(&string) {
             panic!("`{:?}` is not a valid identifier", string)
         }
         if is_raw {
-            let normalized_sym = Symbol::intern(string);
+            let normalized_sym = sym.interned();
```

Comment:
Re-interning here was intentional (to get rid of possible gensyms before checking).

Reply:
Hence the `interned` call.